Raise exception on non-zero return of Inspection (fix #63)

lukpueh opened the pull request:
Fixes https://github.com/in-toto/in-toto/issues/63

- Adds internal function that checks return values of shell commands, e.g. from inspections. Raises exception if the passed value is non-int and non-zero.
- Adds call to function in `verifylib.run_all_inspections`, i.e. verification aborts if a ran inspection returns non-zero.
- Fixes typos in comments
- Adds unittests

Commit:
Adds BadReturnValueError exception

Indicates that a ran command e.g. from an inspection exited with
non-int or non-zero return value.

Commit:
Adds internal return value check function

Adds internal function that checks return values of shell
commands, e.g. from inspections. Raises exception if the passed
value is non-int and non-zero.

Adds call to function in verifylib.run_all_inspections, i.e.
verification aborts if a ran inspection returns non-zero.

Fixes typos in comments

Commit:
Adds return value checker unittests

SantiagoTorres commented on in_toto/verifylib.py:
```diff
@@ -37,10 +37,49 @@
 from in_toto.matchrule_validators import check_matchrule_syntax
 import in_toto.log as log
 
+def _raise_on_bad_retval(return_value, command=None):
```
Just out of curiosity. This private method does not require documentation does it? (Not that I mind)

lukpueh replied:
According to our code-style-guidelines its not required. But I guess it does not hurt either.

SantiagoTorres commented on in_toto/verifylib.py:
```diff
@@ -37,10 +37,49 @@
+    raise BadReturnValueError(msg.format(what="int"))
+
+  # TODO: in-toto specification suggests special behavior on
+  # return_value < 127, but does not fully define that behavior yet
```
I think this comment means to say "return value == 127"

Commit:
Fixes retval comment in verifylib

The specification suggests special behavior on retval==127
but that behavior is not yet fully defined.

SantiagoTorres commented:
Thanks! 👍
